File: python/activities.py
from temporalio import activity

from shared import GreetingInput, MathInput, DoubleInput


@activity.defn
async def compose_greeting(inp: GreetingInput) -> str:
    # >>> BREAKPOINT (activity) <<<
    return f"Hello, {inp.name}!"


@activity.defn
async def compose_farewell(inp: GreetingInput) -> str:
    # >>> BREAKPOINT (activity) <<<
    return f"Goodbye, {inp.name}!"


@activity.defn
async def add(inp: MathInput) -> int:
    # >>> BREAKPOINT (activity) <<<
    return inp.a + inp.b


@activity.defn
async def double(inp: DoubleInput) -> int:
    # >>> BREAKPOINT (activity) <<<
    return inp.value * 2

File: go/activities.go
package hello

import "context"

// Add returns a + b.
func Add(ctx context.Context, in MathInput) (int, error) {
	// >>> BREAKPOINT (activity) <<<
	return in.A + in.B, nil
}

// Double returns value * 2.
func Double(ctx context.Context, in DoubleInput) (int, error) {
	// >>> BREAKPOINT (activity) <<<
	return in.Value * 2, nil
}

// ComposeFarewell returns a farewell string.
func ComposeFarewell(ctx context.Context, in GreetingInput) (string, error) {
	// >>> BREAKPOINT (activity) <<<
	return "Goodbye, " + in.Name + "!", nil
}
